rflx/parser/parser.py

```diff
@@ -214,40 +216,40 @@ def create_message(message: MessageSpec, types: Mapping[str, Type]) -> Message:
             target_node = Field(then.name) if then.name != "null" else FINAL
             if target_node not in field_types.keys() | {FINAL}:
                 raise ParserError(
-                    f'undefined component "{then.name}" in message "{message.full_name}"'
+                    f'undefined component "{then.name}" in message "{message.identifier}"'
                 )
             structure.append(
                 Link(source_node, target_node, then.condition, then.length, then.first)
             )
 
-    return UnprovenMessage(message.full_name, structure, field_types).merged().proven()
+    return UnprovenMessage(message.identifier, structure, field_types).merged().proven()
 
 
-def create_derived_message(derivation: DerivationSpec, messages: Mapping[str, Message],) -> Message:
+def create_derived_message(derivation: DerivationSpec, messages: Mapping[ID, Message]) -> Message:
     base_name = (
-        derivation.base if "." in derivation.base else f"{derivation.package}.{derivation.base}"
+        derivation.base if len(derivation.base.parts) == 2 else derivation.package * derivation.base
```

Comment:
After reading several comparisons like `len(derivation.base.parts) == 2` I wonder whether we should add some method to ID to check whether this is a well-formed name (i.e. package.entity)?

Reply:
That would be nice if we find good generic names.

Did you have the following in mind?
- `package` => `len(self.parts) == 1`
- `entity` => `len(self.parts) == 2`
That would probably fit to our specification, but is not very generic. `ID` is also used for the Ada classes and there that differentiation could be misleading.

I thought also about `top_level` and `second_level` (in the style of top-level-domain, second-level-domain, ...). But that is probably not very intuitive.

Another option would be to add functions instead (e.g. to `common.py`):
- `valid_package_id`
- `valid_type_id`

Reply:
I feared that would spawn another name discussion ;)

While an option, I find the utility functions less preferable than a property of ID. OTOH neither `package/entity` nor `top_level/second_level` are really spot-on.

How about

* `basic` => `len(self.parts) == 1` (possible alternatives: `primitive`)
* `compound` => `len(self.parts) > 1` (possible alternatives: `complex`, `composite`)

Reply:
Although I like the idea, it does not cover the current use cases. We only have checks for exactly one or two parts. To allow more than two parts would make the code wrong.

Reply:
As there is no convincing option, I think it would be the best to keep it as it is.

Reply:
I'm OK with that.